src/geometry/euclidean/ceva_theorem.lean

```diff
@@ -0,0 +1,325 @@
+    rw [hA₁, hB₁],
+    exact sub_self D,
+    simp },
+  { intros a ha,
+    simp only [matrix.head_cons, matrix.cons_val_one, matrix.cons_val_zero] at ha,
+    subst ha,
+    subst hD₁,
+    have hindep := S.ind,
+    rw affine_independent_iff_not_collinear at hindep,
+    apply hindep,
+    have hA₂ : S.points 0 ∈ set.range S.points := by use 0,
+    rw (collinear_iff_of_mem ℝ hA₂),
+    use v₂,
+    intros p hp,
+    rw set.range_comp at hp,
+    simp only [true_and, set.mem_range, set.mem_image, exists_apply_eq_apply] at hp,
+    cases hp with n hpn,
+    fin_cases n,
+    { use 0,
+      rw ← hpn,
+      simp, },
+    { use (r₄ - r₃),
+      rw [← hpn, hB₁, hA₁],
+      simp only [sub_smul, vadd_vadd, ← add_assoc, ← smul_assoc, smul_eq_mul, sub_add_cancel] },
+    { use (r₂ • a - r₁ • a - r₃),
+      rw [← hpn, hC₁, hA₁],
+      simp only [sub_smul, vadd_vadd, ← add_assoc, ← smul_assoc, smul_eq_mul, sub_add_cancel] }}
+end
+
+
+lemma affine_basis_fin3_coord_vsub_smul_sum_eq_zero (O D : V) (S : affine_basis (fin 3) ℝ V)
+  (h₁ : collinear ℝ ({S.points 0, S.points 1, D} : set V))
+  (h₂ : collinear ℝ ({D, O, S.points 2} : set V)) :
+  S.coord 0 O • (D -ᵥ S.points 0) + S.coord 1 O • (D -ᵥ S.points 1) = (0 : V) :=
+begin
+  have h := affine_basis.vsub_eq_coord_smul_sum O S D,
+  have hsub : D -ᵥ O - S.coord 2 O • (D -ᵥ S.points 2) = S.coord 0 O • (D -ᵥ S.points 0) +
+  S.coord 1 O • (D -ᵥ S.points 1),
+  { apply vadd_right_cancel (S.coord 2 O • (D -ᵥ S.points 2)),
+    simp only [vsub_eq_sub, sub_add_cancel, vadd_eq_add],
+    simp only [fin.sum_univ_succ, fin.sum_univ_zero, add_zero] at h,
+    convert h using 1,
+    norm_num,
+    abel },
+  have hO : O ∈ ({D, O, S.points 2} : set V),
+  { simp only [set.mem_insert_iff, true_or, eq_self_iff_true, or_true] },
+  rw (collinear_iff_of_mem ℝ hO) at h₂,
+  cases h₂ with v₁ hv₁,
+  have hD₁ : D ∈ ({S.points 0, S.points 1, D} : set V),
+  { simp only [set.mem_insert_iff, set.mem_singleton, or_true] },
+  rw (collinear_iff_of_mem ℝ hD₁) at h₁,
+  cases h₁ with v₂ hv₂,
+  obtain ⟨r₂, hC₁⟩ := hv₁ (S.points 2)
+    (by simp only [set.mem_insert_iff, set.mem_singleton, or_true]),
+  obtain ⟨r₃, hA₁⟩ := hv₂ (S.points 0)
+    (by simp only [set.mem_insert_iff, true_or, eq_self_iff_true]),
+  obtain ⟨r₄, hB₁⟩ := hv₂ (S.points 1)
+    (by simp only [set.mem_insert_iff, true_or, eq_self_iff_true, or_true]),
+  obtain ⟨r₁, hD₁⟩ := hv₁ D
+    (by simp only [set.mem_insert_iff, true_or, eq_self_iff_true]),
+  simp only [hC₁, hA₁, hB₁, hD₁, vadd_vsub_vadd_cancel_right, vadd_vsub, vsub_vadd_eq_vsub_sub,
+    zero_sub, smul_neg, sub_self] at hsub,
+  rw [hB₁, hA₁, hD₁],
+  simp only [vadd_vsub_vadd_cancel_right, vadd_vsub, vsub_vadd_eq_vsub_sub, zero_sub, smul_neg,
+   sub_self],
+  have hlinind := affine_basis.pair_lin_indep S hA₁ hB₁ hC₁ hD₁,
+  have hv₁ : (r₁ + S.coord 2 O • r₂ - S.coord 2 O • r₁) • v₁ = r₁ • v₁ -
+    S.coord 2 O • (r₁ • v₁ - r₂ • v₁),
+  rw ← sub_smul,
```

Comment:
There are missing `{ ... }` here to focus on single goals. In general, every time a tactic creates multiple goals (like `have hv1 : ...,`) you should put the proof of each goal inside separate `{ ... }` blocks.